kuksa_databroker/databroker/src/broker.rs

```diff
@@ -696,40 +874,49 @@ impl Entries {
         entry_type: EntryType,
         description: String,
         datapoint: Option<Datapoint>,
+        allowed: Option<types::DataValue>,
     ) -> Result<i32, RegistrationError> {
         if let Some(datapoint) = self.get_by_path(&name) {
             // It already exists
             return Ok(datapoint.metadata.id);
         };
 
+        let temp_id = 0;
+
+        let mut new_entry = Entry {
+            metadata: Metadata {
+                id: temp_id,
+                path: name.clone(),
+                data_type,
+                change_type,
+                entry_type,
+                description,
+                allowed,
+            },
+            datapoint: match datapoint {
+                Some(datapoint) => datapoint,
+                None => Datapoint {
+                    ts: SystemTime::now(),
+                    value: DataValue::NotAvailable,
+                },
+            },
+            actuator_target: None,
+        };
+
+        new_entry
+            .validate_allowed_type(&new_entry.metadata.allowed)
+            .map_err(|_err| RegistrationError {})?;
+
         // Get next id (and bump it)
         let id = self.next_id.fetch_add(1, Ordering::SeqCst);
 
         // Map name -> id
         self.path_to_id.insert(name.clone(), id);
 
+        new_entry.metadata.id = id;
```

Comment:
Doesn't pass clippy since this `clone()` isn't needed anymore..

clippy error / warning
```rust
error: redundant clone
   --> kuksa_databroker/databroker/src/broker.rs:914:36
    |
914 |         self.path_to_id.insert(name.clone(), id);
    |                                    ^^^^^^^^ help: remove this
    |
```

```suggestion

        // Get next id (and bump it)
        let id = self.next_id.fetch_add(1, Ordering::SeqCst);

        // Map name -> id
        self.path_to_id.insert(name, id);

        new_entry.metadata.id = id;
```

Reply:
Otherwise, I think this PR looks fine now.